lib/request-fragment.js

```diff
@@ -57,8 +57,17 @@ module.exports = filterHeaders => (
             fragmentRequest.setTimeout(timeout, fragmentRequest.abort);
         }
         fragmentRequest.on('response', response => {
-            if (response.statusCode >= 500) {
-                reject(new Error('Internal Server Error'));
+            const isError500 = response.statusCode >= 500;
+            const isNonPrimaryAndNon200 =
+                (response.statusCode < 200 || response.statusCode >= 300) &&
+                !fragmentAttributes.primary;
+
+            if (isError500 || isNonPrimaryAndNon200) {
+                reject(
+                    new Error(
+                        `Request fragment error. statusCode: ${response.statusCode}; statusMessage: ${response.statusMessage}; url: ${fragmentUrl};`
```

Comment:
Probably, would be better to use JSON here
What is your opinion about it?

Reply:
what do you mean?